Entities/AEmitter.cpp

```diff
@@ -348,12 +344,13 @@ float AEmitter::EstimateImpulse(bool burst)
 
     }
 
-	// Figure out the throttle factor
+	// Scale the emission rate up or down according to the appropriate throttle multiplier.
 	float throttleFactor = 1.0F;
-	if (m_Throttle < 0) {	// Negative throttle, scale down according to the min throttle range
-		throttleFactor += std::abs(m_MinThrottleRange) * m_Throttle;
-	} else if (m_Throttle > 0) {    // Positive throttle, scale up
-		throttleFactor += std::abs(m_MaxThrottleRange) * m_Throttle;
+	float absThrottle = std::abs(m_Throttle);
+	if (m_Throttle < 0) {
+		throttleFactor = throttleFactor * (1 - absThrottle) + (m_NegativeThrottleMultiplier * absThrottle);
```

Comment:
I'm pretty sure you've made this line more complicated than it needs to be. `throttleFactor` is set to 1.0F right above, but you're using it in your math so you effectively have `1.0F * (1 - std::abs(m_Throttle) + (m_WhateverMultiplier * std::abs(m_Throttle))`

Also, can you explain the math changes here a bit, cause I'm pretty sure they're wrong.
Before throttle and throttle multipliers both linearly affected the resulting throttle factor (i.e. emission rate) in a totally straightforward way. Bigger throttle and bigger multiplier meant more emission rate, going from 0 (no emissions at min throttle) to 2 (double emissions at max throttle). Yours doesn't abs the multipliers, meaning that their signs matter (though reasonably they should always be 0 - 1 so that's not a big deal, but probably needs enforcement in the setters), and when I plug the numbers into excel your stuff is super weird (unless I've made a mistake):
![image](https://user-images.githubusercontent.com/7739094/134824306-7afe8371-e536-4fbd-a8fe-be783fda3b88.png)

If you wanna do it yourself, the formulas are:
Old - `=1 + (A2*ABS(B2))`
New - `= 1 * (1-ABS(A2)) + (B2*ABS(A2))`

Reply:
I gave the formulas to Trez in beforehand to be automatically handled by the converter. I guess it's a thing worth putting into the changelog entry even though it'll probably end up confusing modders, especially when this stuff hasn't ever been very apparent in mods.

I thought we discussed the throttle logic change already so I don't really know what you're frothing about.

Reply:
lol it's 100% possible that we did, but also possible that I've forgotten since it was a while back, or that I didn't fully understand.

Still, this seems straight up incorrect to me, like why would having throttle 0.1 and positive throttle multiplier 0.1 result in 0.9 strength emissions? Feel free to shoot me a discord message if you'd rather discuss there.

Reply:
After discussion with 4zK, we're keeping this as-is. There's a followup issue to get rid of these confusing code concepts - #374.